src/stieltjes.jl

```diff
@@ -185,3 +183,199 @@ for Op in (:Hilbert, :StieltjesPoint, :LogKernel, :PowKernel)
+    return Q*PowerLawMatrix(Q,a,t)
+end
+
+####
+# operator generation
+####
+
+# evaluates ∫(t-x)^a Pn(x)P_m(x) dx for the case m=0, i.e. first row of operator, via backwards recursion
+function powerleg_backwardsfirstrow(a::T, t::T, ℓ::Int) where T <: Real
+    ℓ = ℓ+200
+    coeff = zeros(BigFloat,ℓ)
+    coeff[end-1] =  one(BigFloat)
+    for m = reverse(1:ℓ-2)
+        coeff[m] = (coeff[m+2]-t/((a+m+2)/(2*m+1))*normconst_Pnadd1(m,t)*coeff[m+1])/(((a-m+1)/(2*m+1))/((a+m+2)/(2*m+1))*normconst_Pnsub1(m,t))
+    end
+    coeff = PLnorminitial00(t,a)/coeff[1].*coeff
+    return T.(coeff[1:ℓ-200])
+end
+# modify recurrence coefficients to work for normalized Legendre
+normconst_Pnadd1(m::Int, settype::T) where T<:Real = sqrt(2*m+3*one(T))/sqrt(2*m+one(T))
+normconst_Pnsub1(m::Int, settype::T) where T<:Real = sqrt(2*m+3*one(T))/sqrt(2*m-one(T))
+normconst_Pmnmix(n::Int, m::Int, settype::T) where T<:Real = sqrt(2*m+3*one(T))*sqrt(2*n+one(T))/(sqrt(2*m+one(T))*sqrt(2*n-one(T)))
+# useful explicit initial case
+function PLnorminitial00(t::Real, a::Real)
+    return ((t+1)^(a+1)-(t-1)^(a+1))/(2*(a+1))
+end
+
+# compute r-th coefficient of product expansion of order p and order q normalized Legendre polynomials
+productseriescfs(p::T,q::T,r::T) where T = sqrt((2*p+1)*(2*q+1)/((2*(p+q-2*r)+1)))*(2*(p+q-2*r)+1)/(2*(p+q-r)+1)*exp(loggamma(r+one(T)/2)+loggamma(p-r+one(T)/2)+loggamma(q-r+one(T)/2)-loggamma(q-r+one(T))-loggamma(p-r+one(T))-loggamma(r+one(T))-loggamma(q+p-r+one(T)/2)+loggamma(q+p-r+one(T)))/π
+
+# this generates the entire operator via normalized product Legendre decomposition
+# this is very stable but scales rather poorly with high orders, so we only use it for testing
+function productoperator(a::T,t::T,ℓ::Integer) where T
+    op::Matrix{T} = zeros(T,ℓ,ℓ)
+    # first row where n arbitrary and m==0
+    first::Vector{T} = powerleg_backwardsfirstrow(a,t,2*ℓ+1)
+    op[1,:] = first[1:ℓ]
+    # generate remaining rows
+    for p = 1:ℓ-1
+        for q = p:ℓ-1
+            productcfs::Vector{T} = zeros(T,2*ℓ+1)
+            for i = 0:min(p,q)
+                productcfs[1+q+p-2*i] = productseriescfs(p,q,i)
+            end
+            op[p+1,q+1] = dot(first,productcfs)
+        end
+    end
+    # matrix is symmetric
+    for m = 1:ℓ
+        for n = m+1:ℓ
+            op[n,m] = op[m,n]
+        end
+    end
+    return op
+end
+
+# This function returns the full ℓ×ℓ dot product operator, relying on several different methods for first row, second row, diagonal and remaining elements. We don't use this outside of the initial block.
+function gennormalizedpower(a::T, t::T, ℓ::Int) where T <: Real
+    # initialization
+    ℓ = ℓ+3
+    coeff = zeros(T,ℓ,ℓ)
+    # construct first row via stable backwards recurrence
+    first = powerleg_backwardsfirstrow(a,t,2*ℓ+1)
+    coeff[1,:] = first[1:ℓ]
+    # contruct second row via normalized product Legendre decomposition
+    @inbounds for q = 1:ℓ-1
+        productcfs::Vector{T} = zeros(T,2*ℓ+1)
```

Comment:
Why `::Vector{T}`?

Reply:
Not needed anymore, think that was a leftover from originally testing the decomposition.